test/hotspot/jtreg/compiler/lib/compile_framework/CompileFramework.java

```diff
@@ -0,0 +1,153 @@
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package compiler.lib.compile_framework;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+* This is the entry-point for the Compile Framework. Its purpose it to allow
+* compilation and execution of Java and Jasm sources generated at runtime.
+*
+* Please reference the README.md for more explanation.
+*/
+public class CompileFramework {
+    private List<SourceCode> javaSources = new ArrayList<>();
+    private List<SourceCode> jasmSources = new ArrayList<>();
+    private final Path sourceDir = Utils.makeUniqueDir("compile-framework-sources-");
+    private final Path classesDir = Utils.makeUniqueDir("compile-framework-classes-");
+    private ClassLoader classLoader;
+
+    /**
+    * Add a Java source to the compilation.
+    */
+    public void addJavaSourceCode(String className, String code) {
+        javaSources.add(new SourceCode(className, "java", code));
+    }
+
+    /**
+    * Add a Jasm source to the compilation.
+    */
+    public void addJasmSourceCode(String className, String code) {
+        jasmSources.add(new SourceCode(className, "jasm", code));
+    }
+
+    private String sourceCodesAsString(List<SourceCode> sourceCodes) {
+        StringBuilder builder = new StringBuilder();
+        for (SourceCode sourceCode : sourceCodes) {
+            builder.append("SourceCode: ").append(sourceCode.filePathName()).append(System.lineSeparator());
+            builder.append(sourceCode.code()).append(System.lineSeparator());
+        }
+        return builder.toString();
+    }
+
+    /**
+    * Compile all sources: store the sources to the sources directory, compile
+    * Java and Jasm sources and store the generated class-files in the classes
+    * directory.
+    */
+    public void compile() {
+        if (classLoader != null) {
+            throw new CompileFrameworkException("Cannot compile twice!");
+        }
+
+        Utils.printlnVerbose("------------------ CompileFramework: SourceCode -------------------");
+        Utils.printlnVerbose(sourceCodesAsString(jasmSources));
+        Utils.printlnVerbose(sourceCodesAsString(javaSources));
+
+        System.out.println("------------------ CompileFramework: Compilation ------------------");
+        System.out.println("Source directory: " + sourceDir);
+        System.out.println("Classes directory: " + classesDir);
+
+        Compile.compileJasmSources(jasmSources, sourceDir, classesDir);
+        Compile.compileJavaSources(javaSources, sourceDir, classesDir);
```

Comment:
That is why those methods were originally part of the `ComileFramework`, as fields... I think it is easier to pass them as arguments, because that way one can directly and easily understand what are the components that go into the compilation.